File: Python3/프로그래머스/1/42889. 실패율/실패율.py
def solution(N, stages):
    answer = []
    
    reach = len(stages) #해당 스테이지에 도달한 사용자의 수
    n_achive = 0 #해당 스테이지를 클리어하지 못한 사용자의 수
    for i in range(1, N + 1):
        n_achive = sum(1 for c in stages if c == i)
        # print(i, "번 스테이지에 도달했지만 깨지 못한 사용자의 수는", n_achive)
        answer.append([i, n_achive / reach if reach > 0 else 0])
        reach -= n_achive
    
    # print(answer)
    return [x for x, y in sorted(answer, key=lambda x:(-x[1], x[0]))]

File: C++/프로그래머스/1/42889. 실패율/실패율.cpp
#include <algorithm>
#include <vector>
#include <unordered_map>

using namespace std;

vector<int> solution(int N, vector<int> stages) {
    unordered_map<int, float> stored;
    
    int reach = stages.size(); //해당 스테이지에 도달한 사람의 수
    int achieveN; //해당 스테이지를 해결하지 못한 사람의 수
    sort(stages.begin(), stages.end());
    vector<int> answer;
    for(int i = 1; i <= N; i++) {
        answer.push_back(i);
        achieveN = 0;
        for(auto& s : stages) {
            if(s == i) achieveN++;
        }
        
        if(reach > 0)
            stored[i] = achieveN / (float) reach;
        else stored[i] = 0;
        reach -= achieveN;
    }
    
    sort(answer.begin(), answer.end(), [&](int a, int b)->bool {
        if(stored[a] != stored[b])
            return stored[a] > stored[b];
        else return a < b;
    });
    
    return answer;
}